opus/import/populate_obs_instrument_VGUVS_prof.py

```diff
@@ -0,0 +1,586 @@
+
+    return res
+
+def populate_obs_ring_geometry_VGUVS_proj_resolution1_PROF(**kwargs):
+    metadata = kwargs['metadata']
+    index_row = metadata['index_row']
+    res = import_util.safe_column(index_row, 'MINIMUM_RADIAL_RESOLUTION')
+
+    return res
+
+def populate_obs_ring_geometry_VGUVS_proj_resolution2_PROF(**kwargs):
+    metadata = kwargs['metadata']
+    index_row = metadata['index_row']
+    res = import_util.safe_column(index_row, 'MAXIMUM_RADIAL_RESOLUTION')
+
+    return res
+
+def populate_obs_ring_geometry_VGUVS_j2000_longitude1_PROF(**kwargs):
+    return None
+
+def populate_obs_ring_geometry_VGUVS_j2000_longitude2_PROF(**kwargs):
+    return None
+
+def populate_obs_ring_geometry_VGUVS_ring_azimuth_wrt_observer1_PROF(**kwargs):
+    return None
+
+def populate_obs_ring_geometry_VGUVS_ring_azimuth_wrt_observer2_PROF(**kwargs):
+    return None
+
+def populate_obs_ring_geometry_VGUVS_phase1_PROF(**kwargs):
+    return 180.
+
+def populate_obs_ring_geometry_VGUVS_phase2_PROF(**kwargs):
+    return 180.
+
+# Source is star, observer is Voyager
+# Source DELTA SCO is at south, observer Voyager is at north. Target: S ring.
+# Source SIGMA SGR is at south, observer Voyager is at north. This is when
+# start time before 1986-01-24T17:10:13.320. Target: U ring.
+# Source SIGMA SGR is at north, observer Voyager is at south. This is when
+# start time after 1986-01-24T17:10:13.320. Target: N ring.
+# Source IOTA HER is at north, observer Voyager is at south. Target: S ring.
+
+# Incidence angle: the angle between the point where incoming source photons
+# hit the ring, to the north pole of the planet we're looking at (normal vector
+# on the surface of LIT side of the ring, same as source side), always between
+# 0 (parallel to north pole) to 90 (parallel to ring)
+# We would like to use emission angle to get both min/max of incidence angle.
+# Emission angle is 90-180 (dark side), so incidence angle is 180 - emission
+# angle.
+def populate_obs_ring_geometry_VGUVS_incidence1_PROF(**kwargs):
+    metadata = kwargs['metadata']
+    index_row = metadata['index_row']
+    inc = index_row['INCIDENCE_ANGLE']
+
+    max_ea = index_row['MAXIMUM_EMISSION_ANGLE']
+    cal_inc = 180 - max_ea
+    msg = ('The difference between incidence angle and 180 - emission is' +
+           ' more than 0.005, volume: VG_2802.')
+    assert abs(cal_inc - inc) <= 0.005, msg
+    return 180. - max_ea
+
+def populate_obs_ring_geometry_VGUVS_incidence2_PROF(**kwargs):
+    metadata = kwargs['metadata']
+    index_row = metadata['index_row']
+    inc = index_row['INCIDENCE_ANGLE']
+
+    min_ea = index_row['MINIMUM_EMISSION_ANGLE']
+    cal_inc = 180 - min_ea
+    msg = ('The difference between incidence angle and 180 - emission is' +
+           ' more than 0.005, volume: VG_2802.')
+    assert abs(cal_inc - inc) <= 0.005, msg
+    return 180. - min_ea
+
+def _is_voyager_at_north(**kwargs):
```

Comment:
Shouldn't there be some test for the proper datetime here?

Reply:
done